content/en/blog/_posts/2023-10-24-taint-eviction-controller.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: blog
+title: "Kubernetes 1.29: Decouple TaintManager from NodeLifecycleController"
+date: 2023-11-15
+slug: kubernetes-1-29-taint-eviction-controller
+---
+
+**Authors:** Yuan Chen (Apple), Andrea Tosatto (Apple)
+
+This blog discusses a new feature in Kubernetes 1.29 to improve the handling of taint-based pod eviction.
+
+## Background
+
+In Kubernetes 1.29, an  improvement has been introduced to enhance the handling of taint-based pod eviction on nodes. This blog discusses the changes made to the `NodeLifecycleController` to separate its responsibilities and improve overall code maintainability.
+
+### Summary of Changes
+
+The `NodeLifecycleController` previously combined two independent functions:
+
+- Adding a pre-defined set of `NoExecute` taints to nodes based on the node conditions.
+- Performing pod eviction on NoExecute taints
+
+With the latest release, the taint-based eviction implementation has been moved out of the `node-lifecycle-controller` into a separate and independent component called `TaintEvictionController`. This separation aims to disentangle code, enhance code maintainability, and facilitate future extensions to either component.
+
+### How to Use the New Feature?
+
+A new feature gate, `SeparateTaintManager`, has been added. To enable the new feature, users can use the following flags:
+
+- kube-apiserver: `--feature-gates=SeparateTaintManager=true`
+- kube-controller-manager: `--controllers=kube-eviction-controller`
+
+For compatibility, the legacy `TaintManager` can still be used with the `--taint-manager` flag.
```

Comment:
This is not correct. If users want to use the old taint-manager they need to set the feature-flag to false (e.g. `SeparateTaintManager=false`).

What this feature gives the users, is the ability to disable taint-based eviction by setting `SeparateTaintManager=true` and `--controllers=-kube-eviction-controller`.

Reply:
Updated it.